File: rag/src/api/schemas/platform_llm.py
import uuid

from pydantic import BaseModel


class PlatformLLMBasicRead(BaseModel):
    id: uuid.UUID
    display_name: str
    model_name: str
    provider: str
    is_active: bool

    model_config = {"from_attributes": True}


class PlatformLLMRead(PlatformLLMBasicRead):
    requests_used: int
    requests_limit: int

File: web/src/api/types/platform_llm.ts
export type PlatformLLMBasicRead = {
  id: string;
  display_name: string;
  model_name: string;
  provider: string;
  is_active: boolean;
};

export type PlatformLLMRead = PlatformLLMBasicRead & {
  requests_used: number;
  requests_limit: number;
};